6-anthos-install/README.md

```diff
@@ -0,0 +1,353 @@
+# Deploying Bank of Anthos 
+These instructions need to be run on the bastion host. To access the bastion host, open the Google Cloud console, and navigate to Compute Engine in the GKE project. In my instance, `prj-bu1-d-boa-gke` Then, select the `gce-bastion-us-west1-b-01` and click on `ssh`. You need to be a whitelisted member to access this node.
+
+
+## Insall required tools
+```console
+sudo su
+yum install git
+yum install google-cloud-sdk-kpt
+yum install jq -y
+sudo yum install kubectl
+exit
+```
+
+## Pull the repo
+```console
+cd ${HOME}
+git clone https://github.com/GoogleCloudPlatform/terraform-example-foundation-app.git
+```
+
+## Define required environment variables
+When indicated, make sure to replace the values below with the appropriate values based on the outcome of terraform.
+```console
+# replace YOUR_PROJECT_ID with the project id for the project that will host your clusters.
+# For example: prj-bu1-d-boa-gke-ecb0
+export PROJECT_ID=YOUR_PROJECT_ID
+export PROJECT_NUM=$(gcloud projects describe ${PROJECT_ID} --format='value(projectNumber)')
+export CLUSTER_1=gke-boa-us-east1-001
+export CLUSTER_1_REGION=us-east1
+export CLUSTER_2=gke-boa-us-west1-001
+export CLUSTER_2_REGION=us-west1
+export CLUSTER_INGRESS=gke-mci-us-east1-001
+export CLUSTER_INGRESS_REGION=us-east1
+export WORKLOAD_POOL=${PROJECT_ID}.svc.id.goog
+export MESH_ID="proj-${PROJECT_NUM}"
+export ASM_VERSION=1.8
+export ISTIO_VERSION=1.8.3-asm.2
+export ASM_LABEL=asm-183-2
+export CTX_1=$PROJECT_ID_$CLUSTER_1_REGION_CLUSTER_1
+export CTX_2=$PROJECT_ID_$CLUSTER_2_REGION_CLUSTER_2
+export CTX_INGRESS=$PROJECT_ID_$CLUSTER_INGRESS_REGION_CLUSTER_INGRESS
+```
+## Generate Kubeconfig Entries
+In order to install ASM, we need to authenticate to clusters.
+
+```console
+gcloud container clusters get-credentials ${CLUSTER_1} --region ${CLUSTER_1_REGION}
+gcloud container clusters get-credentials ${CLUSTER_2} --region ${CLUSTER_2_REGION}
+gcloud container clusters get-credentials ${CLUSTER_INGRESS} --region ${CLUSTER_INGRESS_REGION}
+```
+## Install ASM
+### Downloading the script
+
+1. Download the version of the script that installs ASM 1.8.3.
+```console
+curl https://storage.googleapis.com/csm-artifacts/asm/install_asm_"${ASM_VERSION}" > install_asm
+```
+2. Make the script executable:
+```console
+chmod +x install_asm
+```
+3. Create a folder to host the installation files and asm packages. This folder will also include `istioctl`, sample apps, and manifests.
+```console
+mkdir -p ${HOME}/asm-${ASM_VERSION} && export PATH=$PATH:$HOME/asm-${ASM_VERSION}
+```
+
+### Install ASM on both clusters
+The following commands run the script for a new installation of ASM on Cluster1 and Cluster2. By default, ASM uses Mesh CA. The `--enable_all` flag will enable the required Google APIs and set IAM permissions.
```

Comment:
we should try to manage infrastructure via Terraform instead of`--enable_all`. APIs and IAM should already be pre provisioned, I assume only the clusterrole/label is required?